tooling/check-commit-messages.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
```

Comment:
This is fairly well done but there's a lot I'd change. I'll push a series of commits to show how I'd do this.